c7n/resources/rds.py

```diff
@@ -175,5 +176,59 @@ def process(self, resources):
                 raise
 
             self.log.info("Deleted rds: %s" % rdb['DBInstanceIdentifier'])
-        
-        
+
+
+@actions.register('snapshot')
+class Snapshots(BaseAction):
+
+    def process(self, resources):
+        with self.executor_factory(max_workers=3) as w:
+            futures = []
+            for resource in resources:
+                futures.append(w.submit(
+                    self.process_rds_snapshot,
+                    resource))
+                for f in as_completed(futures):
+                    if f.exception():
+                        self.log.error(
+                            "Exception creating rds snapshot  \n %s" % (
+                                f.exception()))
+
+    def process_rds_snapshot(self, resource):
+        c = local_session(self.manager.session_factory).client('rds')
+        c.create_db_snapshot(
+            DBSnapshotIdentifier="BKUP-%s-%s" % (
+                resource['DBInstanceIdentifier'],
+                resource['Engine']),
+            DBInstanceIdentifier=resource['DBInstanceIdentifier'])
+
+
+@actions.register('retention')
+class RetentionWindow(BaseAction):
+
+    def process(self, resources):
+        with self.executor_factory(max_workers=3) as w:
+            futures = []
+            for resource in resources:
+                futures.append(w.submit(
+                    self.process_rds_snapshot,
+                    resource))
+                for f in as_completed(futures):
+                    if f.exception():
+                        self.log.error(
+                            "Exception creating rds snapshot  \n %s" % (
+                                f.exception()))
+
+    def process_snapshot_retention(self, resource):
+        try:
+            v = int(resource['BackupRetentionPeriod'])
+            if v == 0:
+                    self.retention_window(resource)
+        except ValueError:
+                self.set_retention_window(resource)
+
+    def set_retention_window(self, resource):
+        c = local_session(self.manager.session_factory).client('rds')
+        c.modify_db_instance(
+            DBInstanceIdentifier=resource['DBInstanceIdentifier'],
+            BackupRetentionPeriod=7)
```

Comment:
this should be configurable source, from self.data['days']

also these need schema attributes to be valid from config